### PR TITLE
usage sms per country code count

### PR DESCRIPTION
**Pull request by shimonewman**

### CODE REVIEW

File: src/Appwrite/Platform/Workers/Messaging.php [hunk 1]
```diff
@@ -119,6 +120,18 @@ public function action(Message $message, Usage $queueForUsage): void
...
+            $phoneUtil = PhoneNumberUtil::getInstance();
+
+            try {
+                $countryCode = $phoneUtil
+                    ->parse($payload['recipient'])
+                    ->getCountryCode();
+
+                $queueForUsage->addMetric(str_replace('{countryCode}', $countryCode, METRIC_MESSAGES_COUNTRY_CODE), 1);
+            } catch (\libphonenumber\NumberParseException $e) {
+                console::error("Error parsing number: " . $e->getMessage());
```
Comment (christyjacob4) [inline src/Appwrite/Platform/Workers/Messaging.php:132]:
```suggestion
                Console::error("Error parsing number: " . $e->getMessage());
```

File: src/Appwrite/Platform/Workers/Messaging.php [hunk 2]
```diff
@@ -119,6 +120,18 @@ public function action(Message $message, Usage $queueForUsage): void
...
 
+            $phoneUtil = PhoneNumberUtil::getInstance();
+
+            try {
+                $countryCode = $phoneUtil
+                    ->parse($payload['recipient'])
+                    ->getCountryCode();
+
+                $queueForUsage->addMetric(str_replace('{countryCode}', $countryCode, METRIC_MESSAGES_COUNTRY_CODE), 1);
+            } catch (\libphonenumber\NumberParseException $e) {
```
Comment (christyjacob4) [inline src/Appwrite/Platform/Workers/Messaging.php:131]: lets import the exception and use `NumberParseException` directly

File: app/init.php [hunk 3]
```diff
@@ -196,6 +196,7 @@
 const METRIC_TEAMS = 'teams';
 const METRIC_USERS = 'users';
 const METRIC_MESSAGES  = 'messages';
+const METRIC_MESSAGES_COUNTRY_CODE  = '{countryCode}.messages';
```
Comment (eldadfux) [inline app/init.php:199]:
```suggestion
const METRIC_MESSAGES_COUNTRY  = '{countryCode}.messages';
```

Code can be the value, we don't need as a prefix.

File: src/Appwrite/Platform/Workers/Messaging.php [hunk 4]
```diff
@@ -17,6 +18,7 @@
 use Utopia\Platform\Action;
 use Utopia\Queue\Message;
 use Appwrite\Event\Usage;
+use libphonenumber\PhoneNumberUtil;
```
Comment (eldadfux) [inline src/Appwrite/Platform/Workers/Messaging.php:21]: Let's not use the lib directly in Appwrite and instead abstract it in utopia/messaging.

**Review by eldadfux — CHANGES_REQUESTED** [review]

**Review by christyjacob4 — APPROVED** [review]
Lets resolve the conflicts and merge 👍

**Pull request merged**